lib/sentry/integrations/check_in_id_mappings.ex

```diff
@@ -5,30 +5,58 @@ defmodule Sentry.Integrations.CheckInIDMappings do
   alias Sentry.UUID
 
   @table :sentry_cron_mappings
+  @sweep_interval_millisec 30_000
 
   @spec start_link(keyword()) :: GenServer.on_start()
-  def start_link([] = _opts) do
-    GenServer.start_link(__MODULE__, nil, name: __MODULE__)
+  def start_link(opts) do
+    GenServer.start_link(__MODULE__, opts, name: __MODULE__)
   end
 
   @spec lookup_or_insert_new(String.t()) :: UUID.t()
-  def lookup_or_insert_new(key) do
-    case :ets.lookup(@table, key) do
-      [{^key, value}] ->
-        value
+  def lookup_or_insert_new(cron_key) do
+    inserted_at = System.system_time(:millisecond)
+
+    case :ets.lookup(@table, cron_key) do
+      [{^cron_key, uuid, _inserted_at}] ->
+        uuid
 
       [] ->
-        value = UUID.uuid4_hex()
-        :ets.insert(@table, {key, value})
-        value
+        uuid = UUID.uuid4_hex()
+        :ets.insert(@table, {cron_key, uuid, inserted_at})
+        uuid
     end
   end
 
   ## Callbacks
 
   @impl true
-  def init(nil) do
+  def init(state) do
     _table = :ets.new(@table, [:named_table, :public, :set])
+    schedule_sweep(state)
     {:ok, :no_state}
   end
+
+  @impl true
+  def handle_info({:sweep, ttl_millisec}, state) do
+    now = System.system_time(:millisecond)
+
+    # All rows (which are {cron_key, uuid, inserted_at}) with an inserted_at older than
+    # now - @ttl_millisec.
+    match_spec = [{{:"$1", :"$2", :"$3"}, [], [{:<, :"$3", now - ttl_millisec}]}]
+    _ = :ets.select_delete(@table, match_spec)
+
+    schedule_sweep(state)
+    {:noreply, state}
+  end
+
+  ## Helpers
+
+  defp schedule_sweep(ttl_millisec) do
+    Process.send_after(self(), {:sweep, ttl_millisec}, @sweep_interval_millisec)
```

Comment:
Do you need to send over the `ttl_millisec` if it's the state of the GenServer anyway?

Reply:
Nope! Thx for pointing that out!

Reply:
Okay by removing this meant I had to make this genserver configurable for testing since we're testing both a singleton and isolated process. So this caused some extra changes in the code